lib/hmmlearn/base.py

```diff
@@ -586,41 +553,18 @@ def _get_n_fit_scalars_per_param(self):
         for a non-degenerate fit.
         """
 
-    def _init(self, X):
+    def _init(self, X, lengths=None):
```

Comment:
You should just normalize `lengths` early (at the only call site, in fit()), and let implementations assume that it is always passed and never None.